mate-panel/panel-run-dialog.c

```diff
@@ -106,13 +106,37 @@ typedef struct {
 enum {
 	COLUMN_GICON,
 	COLUMN_NAME,
+	COLUMN_ACCELERATOR_MASK,
+	COLUMN_ACCELERATOR_KEY_VALUE,
 	COLUMN_COMMENT,
 	COLUMN_PATH,
 	COLUMN_EXEC,
 	COLUMN_VISIBLE,
 	NUM_COLUMNS
 };
 
+typedef struct
+{
+        gint list_item_idx;
+        GdkModifierType modifier;
+        guint key_id;
+} AcceleratorKeyMapping;
+
+const AcceleratorKeyMapping accelerator_key_mapping[] =
+    {
+        {0, GDK_MOD1_MASK, GDK_KEY_1 },
+        {1, GDK_MOD1_MASK, GDK_KEY_2 },
+        {2, GDK_MOD1_MASK, GDK_KEY_3 },
+        {3, GDK_MOD1_MASK, GDK_KEY_4 },
+        {4, GDK_MOD1_MASK, GDK_KEY_5 },
+        {5, GDK_MOD1_MASK, GDK_KEY_6 },
+        {6, GDK_MOD1_MASK, GDK_KEY_7 },
+        {7, GDK_MOD1_MASK, GDK_KEY_8 },
+        {8, GDK_MOD1_MASK, GDK_KEY_9 },
+        {9, GDK_MOD1_MASK, GDK_KEY_0 },
+        {-1, GDK_MOD1_MASK, 0 }
```

Comment:
You could remove this last entry, and use `G_N_ELEMENTS(accelerator_key_mapping)` to know the array length.

Reply:
Fixed